docs/testing_servers.md

```diff
@@ -0,0 +1,340 @@
+# Testing Servers
+
+Testing servers with the conformance runner involves the following steps:
+
+* Implementing the ConformanceService endpoints.
+* Making an executable file that will read messages from `stdin` and write messages to `stdout`.
+* Defining any configuration for what your server supports. For more information on how to do this, see the [Configuration Files](./configuring_and_running_tests.md#configuration-files) section in the [Configuring and Running Tests documentation](./configuring_and_running_tests.md).
+
+## Process
+
+The basic process for working with servers in the conformance suite will work as follows:
+
+* The conformance runner will pass a `ServerCompatRequest` message via `stdin`. This message will contain all the details necessary
```

Comment:
We need a little more clarity on what it means to "pass a message via stdin". In particular, this should describe how the program will need to read four bytes for the 32-bit (network byte order) size of the message, then read that many bytes, and then de-serialize those bytes into this message type.

Same for the "write a message to stdout" below.